SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCQueues.h

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/RPCs/RPCTypes.h"
+
+namespace SpatialGDK
+{
+/**
+ * Unbounded queue.
+ * Will try to flush everything each frame, and keep the Items that could not be flushed for later
+ */
+template <typename Payload, typename AdditionalSendingData = RPCEmptyData>
+struct TRPCUnboundedQueue : public TRPCQueue<Payload, AdditionalSendingData>
+{
+	TRPCUnboundedQueue(FName InName, TRPCBufferSender<Payload>& Sender)
+		: TRPCQueue<Payload, AdditionalSendingData>(InName, Sender)
+	{
+	}
+
+	void FlushAll(RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&) override;
+	void Flush(Worker_EntityId EntityId, RPCWritingContext& Ctx, const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&,
+			   bool bIgnoreAdded = false) override;
+	void OnAuthLost(Worker_EntityId EntityId) override;
+	void OnAuthGained_ReadComponent(const RPCReadingContext& iCtx) override;
+
+protected:
+	bool FlushQueue(Worker_EntityId EntityId, typename TRPCQueue<Payload, AdditionalSendingData>::QueueData& Queue, RPCWritingContext& Ctx,
+					const typename TWrappedRPCQueue<AdditionalSendingData>::SentRPCCallback&);
+};
+
+/**
+ * Queue with a fixed capacity.
+ * It will queue up to the specified capacity and drop additional RPCs
+ * It will drop all Items that were not sent when flushing.
+ */
+template <typename Payload, typename AdditionalSendingData = RPCEmptyData>
+struct TRPCFixedCapacityQueue : public TRPCUnboundedQueue<Payload, AdditionalSendingData>
```

Comment:
Why extend this from the unbounded queue? Fixed/unbounded seem at odds with each other.

Reply:
True I could refactor that a bit.